src/igtcloud/client/tools/upload_project.py

```diff
@@ -32,6 +34,32 @@ def upload_project(local_folder: str, project_name: str, institute_name: str = N
     if submit:
         _password = getpass("For electronic record state it is required to reenter the password")
 
+    if category.lower() == "annotation":
+        # Annotation file upload
+        logger.info("Annotation File Uploading...")
+        if os.path.isdir(local_folder):
+            for root, dirs, local_files in os.walk(local_folder):
+                key = root
+                annotation_file = local_files
+        else:
+            logger.error("Not a valid Directory")
+        key = key + "/" + str(annotation_file[0])
```

Comment:
You only process the first item in the `local_files` list. I think all files should be processed, right?

Reply:
As discussed via Teams, please leverage the threaded upload approach using the `ThreadPoolExecutor`, which is already there in the `upload_project.py` file to allow for multithreaded uploads, instead of doing this sequentially. 

If you add `f_extend=lambda x: x` to the `FilesCollectionWrapper` in the `get_annotation_files` function in `entities_service.py`, you add the ability to call `study.annotations.upload()`. This is the method that you want to pass to the `ThreadPoolExecutor`, e.g. `future = executor.submit(project.files.upload, file_path, key, callback=pbar.update)`.

Reply:
Modified the changes as suggested